File: trunk/activeobject/src/active_domain_name_registry.rb
require "src/domain_name_registry"
require "src/scheduler"
require "src/future_result"

class ActiveDomainNameRegistry
  def initialize(domain_space)
    @domain_name_registiry = DomainNameRegistry.new(domain_space)
    @scheduler = Scheduler.new
  end
  def buy_domain_name(domain_name)
    result = FutureResult.new
    @scheduler.enqueue {
      result.value = @domain_name_registiry.buy_domain_name(domain_name)
    }
    return result
  end
end

File: activeobject/java/src/activeobject/ActiveDomainNameRegistry.java
package activeobject;


public class ActiveDomainNameRegistry {
	private DomainNameRegistry domainNameRegistry;
	private Scheduler scheduler;
	
	public ActiveDomainNameRegistry(String domainSpace) {
		domainNameRegistry = new DomainNameRegistry(domainSpace);
		scheduler = new Scheduler();
	}
	
	public PurchaseResult buyDomainName(final String domainName)
	{
		final FuturePurchaseResult futurePurchaseResult = new FuturePurchaseResult();
		scheduler.enqueue(new Runnable() {
			public void run() {
				futurePurchaseResult.setValue(domainNameRegistry.buyDomainName(domainName));
			}
		});
		return futurePurchaseResult;
	}
}
